editor/src/components/canvas/canvas-strategies/shared-absolute-move-strategy-helpers.ts

```diff
@@ -82,7 +89,28 @@ function createMoveCommandsForElement(
           true,
         )
       } else {
-        return null
+        // if not defined we still try to fix it by adding left and/or top with initial value 0
+        if (pin === 'left' && !isHorizontallyDefined) {
+          return adjustCssLengthProperty(
+            'permanent',
+            selectedElement,
+            stylePropPathMappingFn('left', ['style']),
+            drag.x * (negative ? -1 : 1),
+            elementParentBounds?.width,
+            true,
+          )
+        } else if (pin === 'top' && !isVerticallyDefined) {
+          return adjustCssLengthProperty(
+            'permanent',
+            selectedElement,
+            stylePropPathMappingFn('top', ['style']),
+            drag.y,
+            elementParentBounds?.height,
+            true,
+          )
+        } else {
+          return null
+        }
       }
     },
     ['top', 'bottom', 'left', 'right'] as const,
```

Comment:
Eni's https://github.com/concrete-utopia/utopia/pull/2326 has a slightly different approach to the same problem (adding missing props).

I think it'd make this code clearer if instead of getting `['top', 'bottom', 'left', 'right']` here, it'd only get the props to actually work on, and create the array of props to work on in a previous step. then while making the array you could ensure there's at least 1 prop per dimension (inserting left or top as needed). 
Kind of like how `ensureAtLeastTwoPinsForEdgePosition` works.

Reply:
Great! I refactored it according to your suggestions.